akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ShardCoordinator.scala

```diff
@@ -689,6 +714,7 @@ abstract class ShardCoordinator(typeName: String, settings: ClusterShardingSetti
   def regionTerminated(ref: ActorRef): Unit =
     if (state.regions.contains(ref)) {
       log.debug("ShardRegion terminated: [{}]", ref)
+      regionTerminationInProgress += ref
```

Comment:
CoordinatedShutdownShardingSpec was failing and that was because all GetShardHome requests are not stashed anymore and then for some race conditions the region receives Terminated at ~ the same time as the coordinator and if a message is triggering GetShardHome request at that time it is possible that the coordinator replies with the old location and the message is delivered to the old non-existing node.

Solved by using regionTerminationInProgress also during the update here, which means that the GetShardHome request is ignored and retried.